core/unit_test/CMakeLists.txt

```diff
@@ -52,18 +59,25 @@ endforeach()
 if(${Cabana_ENABLE_MPI})
   foreach(_device ${CABANA_SUPPORTED_DEVICES})
     if(Cabana_ENABLE_${_device})
+      set(_dir ${CMAKE_CURRENT_BINARY_DIR}/${_device})
+      file(MAKE_DIRECTORY ${_dir})
       foreach(_test CommunicationPlan Distributor Halo)
-        add_executable(${_test}_test_${_device} ${_device}/tst${_test}_${_device}.cpp mpi_unit_test_main.cpp)
-        target_link_libraries(${_test}_test_${_device} cabanacore cabana_core_gtest)
+        set(_file ${_dir}/tst${_test}_${_device}.cpp)
```

Comment:
This looks like the same code as above - could we make a function out of this? Might be useful elsewhere as we add new packages.

Reply:
We could.  Won't fix it here though.

Reply:
works for me

Reply:
Let's do the refactor in another PR.